src/python/gudhi/representations/vector_methods.py

```diff
@@ -790,7 +797,13 @@ def fit(self, X, y=None, sample_weight=None):
 
         measures_concat = np.concatenate(X)
         weights_concat = np.concatenate(sample_weight)
-        self.quantiser.fit(X=measures_concat, sample_weight=weights_concat)
+
+        try:
+            self.quantiser.fit(X=measures_concat, sample_weight=weights_concat)
+        except ValueError as ve:
+            print(f'[Atol] fit returned "{ve}.')
+            return self
```

Comment:
Can you explain the use case? If I call `atol.fit_transform(X)` and the quantiser throws an exception, before this patch, I get that exception, while after the patch, a message is printed and then I get a different exception saying that `centers` does not exist?

Reply:
I agree there is not much point in the context of Gudhi, will revert it.